src/Storages/StorageInMemoryMetadata.cpp

```diff
@@ -79,6 +88,42 @@ void StorageInMemoryMetadata::setComment(const String & comment_)
     comment = comment_;
 }
 
+void StorageInMemoryMetadata::setDefiner(const ASTSQLSecurity & sql_security)
+{
+    if (sql_security.definer)
+        definer = sql_security.definer->toString();
+
+    sql_security_type = sql_security.type;
+}
+
+UUID StorageInMemoryMetadata::getDefinerID(DB::ContextPtr context) const
```

Comment:
Do we need this function in the public interface?

Reply:
`StorageInMemoryMetadata` is a struct, so, unfortunately, yes. But overall, I believe it's completely fine to have this method as a public. It does what it says and potentially can be used elsewhere in the future.